File: examples/sorting-algorithms/metrics.ts
export function extract(result: {
	comparisons?: number;
	swaps?: number;
	executionTimeMs?: number;
}): Record<string, number> {
	return {
		comparisons: result.comparisons ?? 0,
		swaps: result.swaps ?? 0,
		executionTimeMs: result.executionTimeMs ?? 0,
	};
}

File: examples/sorting-algorithms/metrics.py
"""Sorting Algorithms Metrics Extractor."""


def extract(result):
    r = result if isinstance(result, dict) else {}
    return {
        "comparisons": r.get("comparisons", 0),
        "swaps": r.get("swaps", 0),
        "executionTimeMs": r.get("executionTimeMs", 0),
    }
